File: snippets/python/tool-result-message_main.py
# from microsoft.agents.xml.models import ChatMessage, FunctionResultContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create tool result message
# message = ChatMessage(
#     role="tool",
#     message_id="msg-result-1",
#     contents=[
#         FunctionResultContent(
#             call_id="call_abc123",
#             name="get_weather",
#             content='{"temperature": 55, "conditions": "partly cloudy"}'
#         )
#     ]
# )
#
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)
# print(xml_output)

File: docs/snippets/csharp/tool-result-message_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create tool result message
var message = new ToolMessage
{
    MessageId = "msg-result-1",
    Contents = new List<AIContent>
    {
        new FunctionResultContent
        {
            CallId = "call_abc123",
            Name = "get_weather",
            Result = "{\"temperature\": 55, \"conditions\": \"partly cloudy\"}"
        }
    }
};

var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);
Console.WriteLine(xmlOutput);